src/storage/src/store.rs

```diff
@@ -183,10 +183,20 @@ pub trait StateStoreIter: Send + 'static {
 pub struct ReadOptions {
     pub epoch: u64,
     pub table_id: Option<TableId>,
+    pub ttl: Option<u32>,
 }
 
 #[derive(Default, Clone)]
 pub struct WriteOptions {
     pub epoch: u64,
     pub table_id: TableId,
 }
+
+impl ReadOptions {
+    pub fn min_epoch(&self) -> u64 {
+        match self.ttl {
+            Some(ttl_u32) => self.epoch - ttl_u32 as u64,
```

Comment:
only the upper 48bit of the epoch represents physical time so you cannot simpy subtract ttl from epoch to construct min_epoch. See epoch.rs for more details.

Reply:
fix it